### PR TITLE
Remove `pallet-getter` usage from pallet-session

### PR DESCRIPTION
**Pull request by mittal-parth**
As per #3326, removes usage of the `pallet::getter` macro from the `session` pallet. The syntax `StorageItem::<T, I>::get()` should be used instead. 

Also, adds public functions for compatibility.

NOTE: The `./historical` directory has not been modified.

cc @muraca

polkadot address: 5GsLutpKjbzsbTphebs9Uy4YK6gTN47MAaz6njPktidjR5cp

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: prdoc/pr_4972.prdoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR removes the `pallet::getter`s from `pallet-session`.
+      The syntax `StorageItem::<T, I>::get()` should be used instead.
+
+crates:
+  - name: pallet-session
+    bump: minor
```
Comment (kianenigma) [inline prdoc/pr_4972.prdoc:14]:
```suggestion
    bump: minor

```

File: substrate/frame/session/src/lib.rs [hunk 2]
```diff
@@ -607,33 +603,53 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+	/// Public function to access the current set of validators.
+	pub fn validators() -> Vec<T::ValidatorId> {
```
Comment (kianenigma) [inline substrate/frame/session/src/lib.rs:607]: We can already mark these as deprecated, and our codebase will be fine right?

**Review by kianenigma — APPROVED** [review]

File: substrate/frame/session/src/lib.rs (hunk 2 above)
Comment (mittal-parth) [reply substrate/frame/session/src/lib.rs:607]: You mean not using the `validators` function?

It is being used in a lot of places outside of the session pallet like [this](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/babe/src/tests.rs#L435)

Comment (seadanda) [reply substrate/frame/session/src/lib.rs:607]: I would say that deprecating the compatibility functions and changing them within the sdk is in scope for this PR. I had a quick flick through a couple of PRs from the linked issue and others have replaced usage of getters outside the pallet in similar situations.

Comment (mittal-parth) [reply substrate/frame/session/src/lib.rs:607]: It's a little dependent. Even I had two ways on this. Some PRs have changed, some haven't. If this sounds, good I can replace all instances (outside the pallet as well). 
cc: @kianenigma

**Review by bkchr — APPROVED** [review]

**Pull request merged**